### ISSUE TITLE
Use Computed output as an input for respurce

### ISSUE THREAD

**Issue by Hassnain-Alvi** [pulumi/pulumi#9733]
### What happened?

I am not sure if i am understanding it correctly but even using apply function doesn't make it work.
I am creating a composer env and in a later step i need to use computed id of that composer resource to query and add a new nodepool resource to it. but it fails during pulumi up computation and raise NoneType error which is epxected because at that time the composer environment doesn't exist 

### Steps to reproduce

```
composer = gcp.composer.Environment(name,
		name = name,
		region = region
	)
	
env = gcp.composer.get_environment(name=composer.name, region=region, opts=pulumi.ResourceOptions(depends_on=[composer]))
```

### Expected Behavior

getEnvironment function should be computed after successful provision of composer environment

### Actual Behavior

it runs on pulumi up during stack preview and raise following exception:

```
env = gcp.composer.get_environment(name=composer.name, region=region, opts=pulumi.ResourceOptions(depends_on=[composer]))
        configs=__ret__.configs,
    AttributeError: 'NoneType' object has no attribute 'configs'
    error: an unhandled error occurred: Program exited with non-zero exit code: 1
```

### Versions used

v3.33.2

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
@Hassnain-Alvi Thank you for your question.

I think you should use the `_output` variant of the function, `get_environment_output`:

```
env = gcp.composer.get_environment_output(name=composer.name, region=region, opts=pulumi.ResourceOptions(depends_on=[composer]))
```

This function accepts outputs in its args properties. See [this blog post](https://www.pulumi.com/blog/functions-accept-outputs/) that highlights the difference between two function styles.

**Comment by Hassnain-Alvi** [comment]
@mikhailshilkov Yeah but its still the same actually the problem is when i run pulumi up it tries to compute the composer output immediately and returns `AttributeError: 'NoneType' object has no attribute 'configs'`
the composer GKE cluster is not available at the planning part 
Its weird that it tries to show it in pulumi approval step. This step should be executed when composer environment is provisioned

**Comment by mikhailshilkov** [comment]
Then this sounds like #9593.

May I ask why you need to call `get_environment` at all? Is the computed id that you mention not available as `composer.id`?

**Comment by Hassnain-Alvi** [comment]
yes seems like it because its dependent on output from another function
I tried using try catch but it didn't seem to work it still raise exception. is there any workaround for this ?
There should be a function that we can call to query resource existence and try catch mechanism in place
